adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java

```diff
@@ -98,4 +100,47 @@ public Route addRoute(@NotNull final Route newRoute) throws IllegalArgumentExcep
     public Route getRouteById(final String routeId) {
         return routeCollection.get(routeId);
     }
+
+    /**
+     * Add a Transfer representing a row from transfers.txt to this. Return the entity added to the repository if the
+     * uniqueness constraint of route based on composite key from_stop_id and to_stop_id is respected, if this
+     * requirement is not met, returns null.
+     *
+     * @param newTransfer the internal representation of a row from transfers.txt to be added to the repository.
+     * @return the entity added to the repository if the uniqueness constraint of route based on composite key
+     * from_stop_id and to_stop_id is respected, if this requirement is not met, returns null.
+     */
+    @Override
+    public Transfer addTransfer(final Transfer newTransfer) throws IllegalArgumentException {
+        if (newTransfer != null) {
+            if ((transferCollection.containsKey(newTransfer.getFromStopId())) &&
+                    (transferCollection.get(newTransfer.getFromStopId()).containsKey(newTransfer.getToStopId()))) {
+                return null;
+            } else {
+                String fromStopId = newTransfer.getFromStopId();
+                String toStopId = newTransfer.getToStopId();
+                Map<String, Transfer> innerMap = new HashMap<>();
+                innerMap.put(toStopId, newTransfer);
+                transferCollection.put(fromStopId, innerMap);
+                return newTransfer;
+            }
+
+        } else {
+            throw new IllegalArgumentException("Cannot add null transfer to data repository");
+        }
+    }
+
+    /**
+     * Return the Transfer representing a row from transfers.txt related to the composite key provided as parameter
+     *
+     * @param fromStopId first part of the composite key: identifies a stop or station where a connection between
+     *                   routes begins
+     * @param toStopId   second part of the composite key: identifies a stop or station where a connection between
+     *                   routes ends
+     * @return the Transfer representing a row from transfers.txt related to the composite key provided as parameter
```

Comment:
I would suggest adding a more explicit note to this Javadoc saying that directionality of the transfer matters - in other words, querying fromStopId of `1` will only give you records where `1` is in the from_stop_id column, and will not return any records where `1` is in the to_stop_id column but not in the ftom_stop_id column.